### ISSUE TITLE
ui_atlas.zim - license / DamonPS2 / etc

### ISSUE THREAD

**Issue by twinaphex** [hrydgard/ppsspp#10402]
Hi there @hrydgard,

I have asked @mudlord to look into that Damon PS2 thing, he has helped me before in the Hyperkin Retron5 situation to reverse the dynamic libraries they were using on Android. In addition to PCSX2 database files he has also found a file called 'ui_atlas.zim'.

They are using a commercial DRM to obfuscate the main emulator library.

libserial.so most definitely contains portions of PCSX2.

I will share more details with you as they become available. Maybe he can also comment in this thread.

**Comment by mudlord** [comment]
libserial.so contains the bits as people on reddit pointed out before. Logic for image parsing is there.

The DRM in use is Jiagu 360, a DRM which does runtime code encryption/decryption of any binary code. Just like VMProtect/Denuvo. Making it very, very difficult to RE statically without unpacking the binary first. And even then, my Android executable unpacking abilities are elementary at best. (its been years since I touched a major Android protection).

I found carbon copies for the PCXS2 game database as well as its own DB for per game ARM dynarec hacks.

**Comment by brujo5** [comment]
I see this to.

Anyways I tested this emulator in android with snapdragon 835 and I can says is very promising. Almost full speed 50 - 60fps in games like cold fear, fatal frame, God of war 2, super dragon ball z etc.
The only thing is you need to pay for the emulator.

**Comment by Thunder07** [comment]
they have not been able to obscure the serial retrieving function since they have a reference to it in android UI itself. the serial reference all point to PCSX2 code. @gregory38

**Comment by twinaphex** [comment]
@Thunder07 Yeah I agree, somebody from PCSX2 and/or @gregory38 probably needs to get involved as well.

**Comment by Thunder07** [comment]
@twinaphex unless explicitly stated I don't see why that file would also not be GPL, however, the fact that the file is there should draw more attention to the fact that more code is probably used, their intro before the PS2 boot sequence looks like ppsspp background
![!bootsq](https://i.ytimg.com/vi/gYvnB53-d68/hqdefault.jpg)

**Comment by iOS4all** [comment]
Can you please give us Cydia repo of this emu or deb file to could test it on my jailbroken iPad Air 2?
Because the creator of demonps2 wrote for android & iOS.

Thanks

**Comment by spring568** [comment]
Don't worry guys, according to the daemon ps2 page they'll open source parts of it once they have 80 million downloads.  By that time they'd have made nearly half a million dollars reselling stolen pcsx2 code

**Comment by meepingsnesroms** [comment]
I have a feeling this is way more than just pcsx2, they got the dynarec from ppsspp or play! since there is no pcsx2 arm dynarec, they obviously have something to hide by encrypting the librarys, something other emu scammers dont even bother with.(I found the symbols in superGnes/superRetro16 by searching "FX" for the superfx symbols in a hex editor)

This feels like ransomware built with stolen emulators.

When something costs money it gives the illusion of legitimacy and it appears this emulator actually works too, they also mention on there app listing that when it gets so many downloads it will become free, a very stupid strategy to make money.

> About Free Versions and Open Source :
To speed up the development of the DamonPS2 Emulator, we need more money and hire more developers. So, After the paid version downloads more than 50,000, we will provide free version to the player to download. In terms of game ROM compatibility, the paid version and the free version will be the same. But the paid version will continue to own and increase the exclusive features that distinguish it from the free version. After more than 80 million downloads from Google Play's free version, we will try to open source gradually. As developers in China, contribute to the word's console emulator open source community.

If any of that where true they would just open source it now, get free development help and take donations like ppsspp.

You should try wireshark, see if it phones home to an obscure domain.

**Comment by blackman91** [comment]
Quick video of the emulator UI and settings, maybe it can provide some clues to you guys:
https://drive.google.com/file/d/1Bg5ljmEBYtq-Sp2HxKmlWXVWcnSy3CFO/view

Also link to the databases the emulator saves to your internal storage:
https://drive.google.com/file/d/1qnnSI7vaIAIYBySaQqpDX97_s2DE3T-R/view

**Comment by hrydgard** [comment]
Very strange stuff. The main menu is clearly Android native while the pause menu looks very ripped from PPSSPP...

**Comment by brujo5** [comment]
![20171212_112029](https://user-images.githubusercontent.com/4172898/33901300-9f323bd8-df2e-11e7-8253-215df50207f7.jpg)
![20171212_111908](https://user-images.githubusercontent.com/4172898/33901301-9f663c94-df2e-11e7-8df8-e3353826d63d.jpg)
![screenshot_20171212-110513](https://user-images.githubusercontent.com/4172898/33901302-9f9989b4-df2e-11e7-9840-4ea602e7923c.jpg)
![20171212_111928](https://user-images.githubusercontent.com/4172898/33901304-9fe74262-df2e-11e7-9269-4e53987f819b.jpg)

Idk if comes to Lydia.
@iOS4all 
Emulator will require opengl3.1 + soon and faster GPU like adreno 540
Idel  if iPhones support that gles version

**Comment by Thunder07** [comment]
@brujo5 and @iOS4all are you fucking idiots seriously having this conversation here.
we're talking about stolen code which makes this emulator illegal, if proven, that code has to be made public which will benefit EVERYONE in the community and unfortunately you 2 idiots... but you 2 just want to get frisky and want to throw your money at him!!

**Comment by brujo5** [comment]
Calm down man  :/ 
do not talk to me about ethics that you are also looking for benefits

Send your  proof to Google. They will remove emulator. 

Exophase drastic ds emulator never made it public source.
